src/scripts/mod.rs

```diff
@@ -0,0 +1,335 @@
+//! 
+//! If the format is different, the script will be silently discarded and will not run. With the Debug option it's possible to see where it goes wrong.
+
+use anyhow::{anyhow, Result};
+use serde_derive::{Deserialize, Serialize};
+use std::collections::HashSet;
+use std::fs::{self, File};
+use std::io::{self, prelude::*};
+use std::net::IpAddr;
+use std::path::PathBuf;
+use subprocess::{Exec, ExitStatus};
+use text_placeholder::Template;
+use crate::input::ScriptsRequired; 
+
+static DEFAULT: &'static str = r#"tags = ["core_approved", "RustScan", "default"]
+developer = [ "RustScan", "https://github.com/RustScan" ]
+ports_separator = ","
+call_format = "nmap -vvv -p {{port}} {{ip}}"
+"#;
+
+pub fn init_scripts(scripts: ScriptsRequired) -> Result<Vec<ScriptFile>> {
+    let mut scripts_to_run: Vec<ScriptFile> = Vec::new();
+
+    match scripts {
+        ScriptsRequired::None => Ok(scripts_to_run),
+        ScriptsRequired::Default => {
+            let default_script = toml::from_str::<ScriptFile>(&DEFAULT).unwrap();
+            scripts_to_run.push(default_script);
+            Ok(scripts_to_run)
+        }
+        ScriptsRequired::Custom => {
+            let scripts_dir_base = match dirs::home_dir() {
+                Some(dir) => dir,
+                None => return Err(anyhow!("Could not infer scripts path.")),
+            };
+            let script_paths = match find_scripts(scripts_dir_base) {
+                Ok(script_paths) => script_paths,
+                Err(e) => return Err(anyhow!(e)),
+            };
+            debug!("Scripts paths \n{:?}", script_paths);
+
+            let parsed_scripts = parse_scripts(script_paths);
+            debug!("Scripts parsed \n{:?}", parsed_scripts);
+
+            let script_config = match ScriptConfig::read_config() {
+                Ok(script_config) => script_config,
+                Err(e) => return Err(anyhow!(e)),
+            };
+            debug!("Script config \n{:?}", script_config);
+
+            // Only Scripts that contain all the tags found in ScriptConfig will be selected.
+            if script_config.tags.is_some() {
+                let config_hashset: HashSet<String> =
+                    script_config.tags.unwrap().into_iter().collect();
+                for script in &parsed_scripts {
+                    if script.tags.is_some() {
+                        let script_hashset: HashSet<String> =
+                            script.tags.clone().unwrap().into_iter().collect();
+                        if config_hashset.is_subset(&script_hashset) {
+                            scripts_to_run.push(script.to_owned());
+                        }
+                        debug!(
+                            "\nScript tags does not match config tags {:?} {}",
+                            &script_hashset,
+                            script.path.clone().unwrap().display()
+                        );
```

Comment:
I think this is always running despite if there's a match or not.

Reply:
Yes, I forget to wrap it back to the else statement. Good catch, fixed it.